packages/paste-core/components/code-block/src/CodeBlockTabList.tsx

```diff
@@ -17,12 +77,67 @@ export interface CodeBlockTabListProps extends Omit<TabListProps, "aria-label">
 
 export const CodeBlockTabList = React.forwardRef<HTMLDivElement, CodeBlockTabListProps>(
   ({ children, element = "CODE_BLOCK_TAB_LIST", ...props }, ref) => {
+    const tab = React.useContext(TabsContext);
+    const theme = useTheme();
+    //  ref to the scrollable element
+    const scrollableRef = React.useRef<HTMLDivElement>(null);
+    const [scrollShadow, setScrollShadow] = React.useState<"none" | "left" | "right" | "both">("none");
+
+    // Function to handle scroll event
+    const handleScroll = (): void => {
+      if (scrollableRef.current) {
+        // No scrollbar, so no shadow
+        if (scrollableRef.current.clientWidth === scrollableRef.current.scrollWidth) {
+          setScrollShadow("none");
+        }
+        // We're positioned on the left most side, so only show right shadow
+        else if (scrollableRef.current.scrollLeft === 0) {
+          setScrollShadow("right");
+        } else if (
+          // We're positioned on the right most side, so only show left shadow
+          scrollableRef.current.scrollLeft + scrollableRef.current.clientWidth ===
+          scrollableRef.current.scrollWidth
+        ) {
+          setScrollShadow("left");
+        } else {
+          // Show both shadows
+          setScrollShadow("both");
+        }
+      }
+    };
+
     return (
-      <Box paddingX="space70">
-        <TabList {...props} aria-label="label" ref={ref} element={element}>
-          {children}
-        </TabList>
-      </Box>
+      <TabPrimitiveList {...(tab as any)} as={Box} {...props} element={element} ref={ref}>
+        <Box
+          element={`${element}_CHILD_WRAPPER`}
+          marginLeft="space70"
+          position="relative"
+          borderBottomStyle="solid"
+          borderBottomWidth="borderWidth10"
+          borderBottomColor="colorBorderInverseWeaker"
+        >
+          <Box
+            {...safelySpreadBoxProps(props)}
+            as={StyledTabList as any}
+            ref={scrollableRef}
+            display="flex"
+            flexWrap="nowrap"
+            flexShrink={0}
+            element={`${element}_CHILD`}
+            overflowX="auto"
+            overflowY="hidden"
+            onScroll={handleScroll}
+          >
+            {children}
```

Comment:
issue: looks like this change is pushing the tab children themselves up by a pixel or so (see how the top box shadow on selected tabs is being cut off the tiniest bit)

old:
![image](https://github.com/user-attachments/assets/fe078ac5-3a1c-4dc3-a66d-84aceffcbb76)
current:
![image](https://github.com/user-attachments/assets/c96dced9-bf20-4b86-bdae-bef0f70660a2)

Reply:
Fixed in latest commit